proposals/reward-proposal-v2-reward-distribution.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Epoch Reward Distribution
+---
+
+## Problem
+
+In the `Epoch Reward Calculation V2`, we proposed a new method to handle
+calculating epoch rewards. This is the second part of that proposal - `Reward
+Distribution`.
+
+## Proposed Solutions
+
+Similar to [Partitioned Inflationary Rewards
+Distribution](https://github.com/solana-labs/solana/blob/master/docs/src/proposals/partitioned-inflationary-rewards-distribution.md),
+the rewards are distributed over `M` slots.
+
+To address the concern for the lack of double-entry booking for reward
+distribution, a special system instruction, `DistributeReward`, will be
+introduced to distribute the rewards from the family of `EpochRewardReserve`
+accounts and the `EpochRewardProof` system account.
+
+This instruction takes the target reward account, the amount to distribute, the
+`EpochRewardReserve` account, and the merkle proof for the distribution. When
+executing this instruction, it first verifies the supplied merkle proof against
+the proof stored in in `EpochRewardProof`. If the verification passed, it will
+transfer the reward from the `EpochRewardReserve` account to the target reward
+account.
+
+To spread out the reward distribution, for each slot, a fixed number (`K`) of
+stake/vote accounts's rewards are distributed. The leaders, within `M`
+slot-height, will be responsible for distributing the rewards. Ideally, these
+`K` rewards are distributed from independent `EpochRewardReserve` accounts. The
+leader will inject `K` `DistributeReward` instructions as transactions into the
+block and propagate those transaction down to the network. All nodes on the
+network will execute those transactions and distribute the rewards. In this way,
+all reward distribution will be recoded as transactions in the first `M` blocks
+after reward computation.
+
+### Challenges and Difficulties
+
+The following are a few questions and challenges of the `V2` reward
+distribution approach for further discussion.
```

Comment:
Like the other part, when are these issues going to be resolved?
Shouldn't these be resolved as part of the proposal before doing any implementation?